tensorflow_addons/callbacks/tqdm_progress_bar.py

```diff
@@ -93,71 +93,74 @@ def __init__(
-                bar_format=self.epoch_bar_format,
-                leave=self.leave_epoch_progress,
-                dynamic_ncols=True,
-                unit=self.mode,
-            )
-
+    def _initialize_progbar(self, hook, epoch, logs=None):
         self.num_samples_seen = 0
         self.steps_to_update = 0
         self.steps_so_far = 0
         self.logs = defaultdict(float)
-
-    def on_epoch_end(self, epoch, logs={}):
-
-        if self.show_epoch_progress:
-            metrics = self.format_metrics(logs)
-            self.epoch_progress_tqdm.desc = metrics
-
-            # set miniters and mininterval to 0 so last update displays
-            self.epoch_progress_tqdm.miniters = 0
-            self.epoch_progress_tqdm.mininterval = 0
-
-            # update the rest of the steps in epoch progress bar
-            self.epoch_progress_tqdm.update(
-                self.total_steps - self.epoch_progress_tqdm.n
+        self.num_epochs = self.params["epochs"]
+        self.mode = "steps"
+        self.total_steps = self.params["steps"]
+        if hook == "train_overall":
+            if self.show_overall_progress:
+                self.overall_progress_tqdm = self.tqdm(
+                    desc="Training",
+                    total=self.num_epochs,
+                    bar_format=self.overall_bar_format,
+                    leave=self.leave_overall_progress,
+                    dynamic_ncols=True,
+                    unit="epochs",
+                )
+        elif hook == "test":
+            if self.show_epoch_progress:
+                self.epoch_progress_tqdm = self.tqdm(
+                    total=self.total_steps,
+                    desc="Evaluating",
+                    bar_format=self.epoch_bar_format,
+                    leave=self.leave_epoch_progress,
+                    dynamic_ncols=True,
+                    unit=self.mode,
+                )
+        elif hook == "train_epoch":
+            current_epoch_description = "Epoch {epoch}/{num_epochs}".format(
+                epoch=epoch + 1, num_epochs=self.num_epochs
             )
-            self.epoch_progress_tqdm.close()
-
-        if self.show_overall_progress:
-            self.overall_progress_tqdm.update(1)
-
-    def on_batch_end(self, batch, logs={}):
+            if self.show_epoch_progress:
+                print(current_epoch_description)
+                self.epoch_progress_tqdm = self.tqdm(
+                    total=self.total_steps,
+                    bar_format=self.epoch_bar_format,
+                    leave=self.leave_epoch_progress,
+                    dynamic_ncols=True,
+                    unit=self.mode,
+                )
+
+    def _clean_up_progbar(self, hook, logs):
+        if hook == "train_overall":
+            if self.show_overall_progress:
+                self.overall_progress_tqdm.close()
+        elif hook == "test" or hook == "train_epoch":
+            if self.show_epoch_progress:
+                if hook == "test":
```

Comment:
a bit too much nest, you can bring this if/else statement outside this nest like

```
   else:
         if hook == "test":
               metrics = self.format_metrics(logs, self.num_samples_seen)
         else:
               metrics = self.format_metrics(logs)
         if self.show_epoch_progress:
               ......
```

Reply:
@shun-lin , After last commit - I was able to generate progress bar in colab. But still gives error that desc is None in tests. Can you please help?

Reply:
sure! which test cases?

Reply:
[this one](https://github.com/tensorflow/addons/pull/1649/checks?check_run_id=584944056)

Reply:
hmmmm I am thinking it may be `self.logs`'s values become None when we call `self._clean_up_progbar("test", self.logs)` at line 207, can you bring out metrics at line 151 like 

```
                 if hook == "test":  
                  metrics = self.format_metrics(logs, self.num_samples_seen)
                  print(metrics) <---- add this to see if metrics is not messed up.
                else:
                    metrics = self.format_metrics(logs)
```

Reply:
ops I think I find a pre-existing problem, if you look at `_update_progbar` we are updating `self.logs`  in ` if self.steps_so_far < self.total_steps:` this is not too big of a problem when training because total_steps is normally greater than 1 but in eval there are many cases when total steps is 1 and thus this will never be called, may you try to either move `self.steps_so_far += 1` after the following `if` block or change `<` to `<=`? Thanks and I apologize for the bug.

Reply:
great. I was aware of this small detail in the first commit of this PR. But forgot now. 😃

Reply:
I think this should work now.

Reply:
sounds good, thanks!